File: topicos/imperativa/lista1/c/ex4-reverse-list.c
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>

int main() {
  int size;

  printf("Qual o tamanho do array: ");
  scanf("%d", &size);

  int *list = malloc(sizeof(int) * size);

  printf("\nDigite uma lista de numeros separados por espaco e pressione ENTER "
         "quando terminar: ");

  for (int i = 0; i < size; i++) {
    scanf("%d", list + i);
  }

  for (int i = size - 1; i >= 0; i--) {
    printf("%d ", list[i]);
  }

  return 0;
}

File: topicos/imperativa/lista1/go/ex4-reverse-list.go
package main

import "fmt"

func main() {
	var size int

	fmt.Printf("Digite o tamanho da lista: ")
	fmt.Scan(&size)

	list := make([]int, size)

	fmt.Printf("Entre com os elementos da lista separados por espaco e pressione Enter quando terminar: ")
	for i := 0; i < size; i++ {
		fmt.Scan(&list[i])
	}

	for size > 0 {
		fmt.Printf("%d ", list[size-1])
		size--
	}
}
